tests/transforms/test_spectrogram.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import torch
+from torchaudio.transforms import Spectrogram
+
+from ml4gw.transforms import MultiResolutionSpectrogram
+
+
+@pytest.fixture(params=[2, 4, 5])
+def kernel_length(request):
+    return request.param
+
+
+@pytest.fixture(params=[128, 256])
+def sample_rate(request):
+    return request.param
+
+
+@pytest.fixture(params=[1, 10])
+def batch_size(request):
+    return request.param
+
+
+@pytest.fixture(params=[1, 3])
+def num_channels(request):
+    return request.param
+
+
+@pytest.fixture(params=[[64], [64, 128], [64, 128, 256]])
+def n_ffts(request):
+    return request.param
+
+
+@pytest.fixture(params=[[50]])
+def win_lengths(request):
+    return request.param
+
+
+@pytest.fixture(params=[[2], [2, 2, 2, 2]])
+def powers(request):
+    return request.param
+
+
+def test_multi_resolution_spectrogram(
+    kernel_length,
+    sample_rate,
+    batch_size,
+    num_channels,
+    n_ffts,
+    win_lengths,
+    powers,
+):
+
+    # List length of spectrogram parameters must be compatible
+    if len(powers) == 4 and len(n_ffts) > 1:
+        with pytest.raises(ValueError):
+            spectrogram = MultiResolutionSpectrogram(
+                kernel_length,
+                sample_rate,
+                n_fft=n_ffts,
+                win_length=win_lengths,
+                power=powers,
+            )
+        return
+
+    # Creating a MRS without any spectrogram arguments should
+    # just create a single default torchaudio histogram with
+    # `normalized = True`
+    spectrogram = MultiResolutionSpectrogram(kernel_length, sample_rate)
+    with pytest.raises(ValueError):
+        spectrogram(torch.ones([4, 3, kernel_length * sample_rate + 1]))
+
+    X = torch.ones([batch_size, num_channels, kernel_length * sample_rate])
+    y = spectrogram(X)
+    expected_y = Spectrogram(normalized=True)(X)
+
+    assert (y == expected_y).all()
+
+    # The `normalized = False` should be ignored
+    spectrogram = MultiResolutionSpectrogram(
+        kernel_length, sample_rate, normalized=[False]
```

Comment:
I think it would be better to raise a `ValueError` in this instance to let the user know this isn't possible. Otherwise the user will silently get unexpected behavior.

Reply:
Agreed.